docs/docs/how-tos/nebari-upgrade.md

```diff
@@ -6,7 +6,79 @@ description: An overview of how to upgrade Nebari. Includes instructions on upda
+
+
+## Minor upgrade to Nebari
+
+### Backup existing data
+
+Perform manual backups of the NFS data and JupyterHub database.
+
+Always [backup your data](./manual-backup.md) before upgrading!
+
+### Install latest version of Nebari
+
+Use pip or conda to install the latest Nebari version.
+
+```shell
+pip install --upgrade nebari
+```
+or 
+
+```shell
+conda update nebari
+```
+
+
+### Update configuration file
+
+You may be deploying Nebari based on a local configuration file, or you may be using CI/CD workflows in GitHub or GitLab.  Either way, you will need to locate a copy of your `nebari-config.yaml` configuration file to upgrade it (and commit back to your git repo in the CI/CD case).
+ 
+For CI/CD deployments, you will need to `git clone <repo URL>` into a folder on your local machine if you haven't done so already.
+
+In the `nebari-config.yaml` file, update the latest version of Nebari.
+
+```yaml
+nebari_version: <version>
+```
+
+Next, its a good idea to validate your configuration file.  While not required, this is the best practice.
+
+```shell
+run nebari validate -c nebari-config.yaml
+```
+
+### Re-deploy Nebari
+
+If you are deploying Nebari from your local machine (see [this section](#cicd-render-and-commit-to-git) for CI/CD deployments), you will now have a `nebari-config.yaml` file that you can deploy.
+
+```shell
+nebari deploy -c nebari-config.yaml   
+```
+
+At this point you may see an error message saying that deployment is prevented due to the `prevent_deploy` setting in your YAML file. This is a safeguard to ensure that you only proceed if you are aware of possible breaking changes in the current upgrade.
+
+For example, we may be aware that you will lose data due to this upgrade, so need to note a specific upgrade process to keep your data safe. Always check the release notes of the release in this case and get in touch with us if you need assistance. For example, you may find that your existing cluster is intentionally deleted so that a new replacement can
+be deployed instead, in which case your data must be backed up so it can be restored after the upgrade.
+
+**When you are ready, remove the `prevent_deploy` setting from the config file, and run the `nebari deploy -c nebari-config.yaml` again.**
+
+
+### CI/CD: render and commit to git
+
+For CI/CD (GitHub/GitLab) workflows, you will need to regenerate the workflow files based on the latest `Nebari` version's templates.
+
+With the newly upgraded `nebari-config.yaml` file, run:
+
+```shell
+nebari render -c nebari-config.yaml
+```
+
+(Note that `nebari deploy` would perform this render step too, but will also immediately redeploy your Nebari.)
+
+Commit all the files (`nebari-config.yaml` and GitHub/GitLab workflow files) back to the remote repo. All files need to be committed together in the same commit.
+
 
 
 ## Upgrade Nebari from pre-0.4.0 to 0.4.0
```

Comment:
Maybe it's worth making it clear that at the time of the 0.4.0 release, the package was still called `qhub`.

Reply:
This also makes me think we need one additional section which covers upgrading from `qhub` to `nebari`...

I'm sorry for continuing to increase the scope of this PR but the more I think of it, this section below only makes sense if the package is still called `qhub`. Above this section, we can have a brief section called, `## Migrating from QHub to Nebari`. In this section, we would simply outline that given the number of resources that will need to be renamed, it is best to think of this as a migration rather than an upgrade. This is because if the user is running `qhub` and then try to upgrade to `nebari` everything will get deleted and rebuilt anew. See this is [warning message in the release notes](https://github.com/nebari-dev/nebari/blob/develop/RELEASE.md#warning).

cc @kcpevey

Reply:
In this section we can also share that we switched from SemVer to CalVer version systems so the last `qhub` release is `0.4.5` and the first `nebari` release is `2022.10.1`.